src/shared/components/iframe-runtime.tsx

```diff
@@ -14,31 +12,36 @@ interface ILogRequest {
 }
 
 interface IProps {
-  url: string;
-  id: string;
-  authoredState: any;
+  authoredState?: any;
+  id?: string;
+  iframeStyling?: any;
   interactiveState: any;
-  setInteractiveState: (state: any) => void;
-  report?: boolean;
+  logRequestData?: Record<string, unknown>;
   navImageUrl?: string;
   navImageAltText?: string;
+  report?: boolean;
+  scaffoldedQuestionLevel?: number;
+  url?: string;
+  wrapper?: WrapperType;
+  setInteractiveState: (state: any) => void | null;
 }
 
 export const IframeRuntime: React.FC<IProps> =
-  ({ url, id, authoredState, interactiveState, setInteractiveState, report, navImageUrl, navImageAltText }) => {
-  const [ iframeHeight, setIframeHeight ] = useState(300);
-  const [ hint, setHint ] = useState("");
-  const iframeRef = useRef<HTMLIFrameElement>(null);
-  const phoneRef = useRef<IframePhone>();
-  // Why is interativeState and setInteractiveState kept in refs? So it's not necessary to declare these variables as
-  // useEffect's dependencies. Theoretically this useEffect callback is perfectly fine either way, but since
-  // it reloads the iframe each time it's called, it's not a great experience for user when that happens while he is
-  // interacting with the iframe (e.g. typing in textarea). And interactiveState is being updated very often,
-  // as well as setInteractiveState that is generated during each render of the parent component.
-  const interactiveStateRef = useRef<any>(interactiveState);
-  const setInteractiveStateRef = useRef<((state: any) => void)>(setInteractiveState);
-  interactiveStateRef.current = interactiveState;
-  setInteractiveStateRef.current = setInteractiveState;
+  ({ authoredState, id, iframeStyling, interactiveState, logRequestData, report,
```

Comment:
It looks like `navImageUrl` and `naveImageAltText` are not used here, so they could be removed from the props type definition.

Reply:
looks good